Sources/WebSocketKit/WebSocket.swift

```diff
@@ -272,22 +348,59 @@ public final class WebSocket: Sendable {
             } else {
                 self.close(code: .protocolError, promise: nil)
             }
+            
         case .text, .binary:
-            // create a new frame sequence or use existing
-            self.frameSequence.withLockedValue { currentFrameSequence in
-                var frameSequence = currentFrameSequence ?? .init(type: frame.opcode)
-                // append this frame and update the sequence
-                frameSequence.append(frame)
-                currentFrameSequence = frameSequence
+            // is compressed, has pmce configured and enabled ?
+            if frame.rsv1 ,
+               let pmce = pmce,
+                   pmce.enabled {
+               
+                do {
+                    let newFrame:WebSocketFrame
+                    if frame.maskKey != nil {
+                        let unmasked = pmce.unmasked(frame: frame)
+                        newFrame = try pmce.decompressed(unmasked)
+                    }else {
+                        newFrame = try pmce.decompressed(frame)
+                    }
+                    
+                    self.frameSequence.withLockedValue { currentFrameSequence in
+                        var frameSequence = currentFrameSequence ?? .init(type: frame.opcode)
+                        // append this frame and update the sequence
+                        frameSequence.append(newFrame)
+                        currentFrameSequence = frameSequence
+                    }
+                }
+                catch {
+                    logger.error("websocket-kit: \(error)")
+                }
```

Comment:
No.

Reply:
not sure what to do here as this signature also does not throw but there is no promise to fail either.